src/models/transfer.py

```diff
@@ -123,10 +142,10 @@ def amount(self) -> float:
 
     def add_slippage(self, slippage: SolverSlippage, log_saver: PrintStore) -> None:
         """Adds Adjusts Transfer amount by Slippage amount"""
-        assert self.receiver == slippage.solver_address, "receiver != solver"
+        assert self._solver == slippage.solver_address, "receiver != solver"
         adjustment = slippage.amount_wei
         log_saver.print(
-            f"Deducting slippage for solver {self.receiver}"
+            f"Deducting slippage for solver {self._solver}"
```

Comment:
Shouldn't these two use the accessor instead of the private field?

Reply:
Well... slippages are returned from the dune query with solver addresses. If we were to use the recipient property here, then we would run the risk of adding slippage having to come before redirects are applied (which is the case anyway). This mean, yes we could use recipient and the code would function as intended, however if someone were to apply redirects first for some reason and then attempt to apply slippage, the program would not act as intended. 

Its fine with me if you insist that we change this, but then we are relying on unit tests to capture potential logical mistakes. You just let me know how you would prefer to proceed, but I think this is good left as is.

It was my impression that it is not unusual to access private fields internally within the class.